### PR TITLE
Cache Hook when initializing `CloudFormationCreateStackSensor`

### PR DESCRIPTION
**Pull request by kaxil**
- `CloudFormationCreateStackSensor` created `AWSCloudFormationHook` in its init method
- This commit delays creation of hook until it is needed

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).

### CODE REVIEW

**Review by eladkal — APPROVED** [review]

File: airflow/providers/amazon/aws/sensors/cloud_formation.py [hunk 1]
```diff
@@ -18,6 +18,11 @@
 """This module contains sensors for AWS CloudFormation."""
 from typing import Optional
 
+try:
+    from functools import cached_property
+except ImportError:
+    from cached_property import cached_property
```
Comment (turbaszek) [inline airflow/providers/amazon/aws/sensors/cloud_formation.py:24]: Is there a reason why we need both imports?

Comment (turbaszek) [reply airflow/providers/amazon/aws/sensors/cloud_formation.py:24]: If this is an issue then we probably should fix all other operators that use `from cached_propert`, WDYT?

Comment (eladkal) [reply airflow/providers/amazon/aws/sensors/cloud_formation.py:24]: We do, see https://github.com/apache/airflow/pull/14606

Comment (kaxil) [reply airflow/providers/amazon/aws/sensors/cloud_formation.py:24]: Yup, like Elad mentioned  #14606 removed the dep for Python 3.8 users. this just build on top of that

**Pull request merged**